src/coreclr/tools/aot/ILCompiler/ILCompiler.csproj

```diff
@@ -15,10 +15,10 @@
     <PublishReadyToRun Condition="'$(UseNativeAotForComponents)' != 'true'">true</PublishReadyToRun>
     <PublishSingleFile Condition="'$(UseNativeAotForComponents)' != 'true'">true</PublishSingleFile>
     <PublishTrimmed Condition="'$(UseNativeAotForComponents)' != 'true'">true</PublishTrimmed>
-    <SuppressGenerateILCompilerExplicitPackageReferenceWarning>true</SuppressGenerateILCompilerExplicitPackageReferenceWarning>
   </PropertyGroup>
 
-  <ItemGroup Condition="'$(UseNativeAotForComponents)' == 'true'">
+  <ItemGroup Condition="'$(UseNativeAotForComponents)' == 'true' and '(MicrosoftDotNetILCompilerVersion)' != ''">
+    <SuppressGenerateILCompilerExplicitPackageReferenceWarning>true</SuppressGenerateILCompilerExplicitPackageReferenceWarning>
```

Comment:
SuppressGenerateILCompilerExplicitPackageReferenceWarning is a property, but this is ItemGroup